compiler-plugin/src/main/kotlin/com/google/devtools/ksp/KotlinSymbolProcessingExtension.kt

```diff
@@ -189,7 +204,7 @@ abstract class AbstractKotlinSymbolProcessingExtension(val options: KspOptions,
         if (finished) {
             logger.reportAll()
         }
-        return AnalysisResult.EMPTY
+        return if (logger.hasError()) AnalysisResult.compilationError(BindingContext.EMPTY) else AnalysisResult.EMPTY
```

Comment:
`analysisCompleted` needs to be updated or removed (and move its logic here) as well.